src/main/java/practice/CandidateValidator.java

```diff
@@ -1,5 +1,26 @@
 package practice;
 
-public class CandidateValidator {
-    //write your code here
+import java.util.function.Predicate;
+import model.Candidate;
+
+public class CandidateValidator implements Predicate<Candidate> {
+    private static final int MIN_AGE = 35;
+    private static final String TARGET_NATIONALITY = "Ukrainian";
+    private static final int MIN_YEARS_PERIOD = 10;
+
+    @Override
+    public boolean test(Candidate candidate) {
+
+        boolean isValidAge = candidate.getAge() >= MIN_AGE;
```

Comment:
don't need to create local variables here
```suggestion
        return candidate.getAge() >= MIN_AGE
                && candidate.isAllowedToVote()
                && candidate.getNationality().equals(TARGET_NATIONALITY)
                && checkTimeLivingInCountry(candidate);
```